test/commands/group_test.exs

```diff
@@ -1,176 +1,245 @@
-  use Cog.AdapterCase, adapter: "test"
+  use Cog.CommandCase, command_module: Cog.Commands.Group
 
-  @moduletag :skip
+  import Cog.Support.ModelUtilities, only: [user: 1,
+                                            group: 1,
+                                            role: 1,
+                                            add_to_group: 2]
+  describe "group subcommands" do
 
-  alias Cog.Models.Group
-  alias Cog.Repository.Groups
+    test "can't pass an unknown subcommand" do
+      {:error, error} =
+        new_req(args: ["not-a-subcommand"])
+        |> send_req()
 
-  setup do
-    user = user("belf", first_name: "Buddy", last_name: "Elf")
-    |> with_chat_handle_for("test")
-    |> with_permission("operable:manage_users")
-    |> with_permission("operable:manage_groups")
+      assert(error == "Unknown subcommand 'not-a-subcommand'")
+    end
 
-    {:ok, %{user: user}}
   end
 
-  test "adding and removing users to groups", %{user: user} do
-    response = send_message(user, "@bot: operable:group member add elves #{user.username}")
-    assert_error_message_contains(response, "Could not find 'user group' with the name 'elves'")
+  describe "group CRUD" do
 
-    [response] = send_message(user, "@bot: operable:group create elves")
-    assert response.name == "elves"
+    test "can be created" do
+      {:ok, response} =
+        new_req(args: ["create", "created_group"])
+        |> send_req()
 
-    response = send_message(user, "@bot: operable:group member add elves papa_elf")
-    assert_error_message_contains(response, "Could not find 'user' with the name 'papa_elf'")
+      assert(%{name: "created_group"} = response)
+    end
 
-    [response] = send_message(user, "@bot: operable:group member add elves")
-    assert response.error == "Missing required args. At a minimum you must include the user group and at least one user name to add"
+    test "can be deleted" do
+      group("deleted_group")
 
-    [response] = send_message(user, "@bot: operable:group member add elves belf")
-    member = hd(response.members)
-    assert member.username == "belf"
+      {:ok, response} =
+        new_req(args: ["delete", "deleted_group"])
+        |> send_req()
 
-    [response] = send_message(user, "@bot: operable:group member remove elves belf")
-    assert length(response.members) == 0
-  end
+      assert(%{name: "deleted_group"} = response)
+    end
 
-  test "adding and removing roles to groups", %{user: user} do
-    group("cheer")
-    role("admin")
+    test "can be renamed" do
+      group("renamed_group")
 
-    [response] = send_message(user, "@bot: operable:group role add cheer admin")
-    assert response.name == "cheer"
+      {:ok, response} =
+        new_req(args: ["rename", "renamed_group", "has_been_renamed"])
+        |> send_req()
 
-    [response] = send_message(user, "@bot: operable:group info cheer")
-    assert hd(response.roles).name == "admin"
+      assert(%{name: "has_been_renamed"} = response)
```

Comment:
Nope `=` is correct. There is actually more to the record than just the name. There is also an id, which we don't care about, and `members` and `roles` keys that are empty.

Actually as I think about it, I should have included the members and roles to confirm that they are empty.